syntax = "proto3";

package auth.v1;

option go_package = "gateway/proto/auth/v1;authv1";

import "common/v1/common.proto";

service AuthService {
  rpc Login (LoginRequest) returns (LoginResponse) {};
  rpc Register (RegisterRequest) returns (RegisterResponse) {};
}

message LoginRequest {
  string email = 1;
  string password = 2;
}

message LoginResponse {
  bool success = 1;
  common.v1.Uuid user_id = 2;
  common.v1.Error error = 3;
}

message RegisterRequest {
    string name = 1;
    string email = 2;
    string password = 3;
}
message RegisterResponse {
    bool success = 1;
    string error = 2;
    uint32 error_code = 3; 
}